src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -842,7 +842,7 @@ def addNodes(self, nodeTypes: Set[str], numNodes, preemptable, spotBid=None) ->
             subnet_id = next(iter(self._worker_subnets_by_zone[zone]))
 
         keyPath = self._sseKey if self._sseKey else None
-        userData = self._getIgnitionUserData('worker', keyPath, preemptable)
+        userData = self._getIgnitionUserData('worker', keyPath, preemptable, self._architecture)
```

Comment:
This could use `type_info.architecture` which will always be defined. Or you could default this when undefined, like you are doing in `_discoverAMI`.

But maybe what you really should do is bring back `getArchitecture()`, and have it return `self._architecture` if it is set, or the current machine's architecture otherwise, and then call it everywhere you need an architecture to use? Then you won't need to repeat the same try/catch everywhere.

Or, you could set `self._architecture` to the current machine's architecture in `__init__`, so we default to assuming the same architecture as we're running on unless someone specifies differently in a `launchCluster()` call.